libs/agent-sm/agent/src/test/java/org/opensearch/javaagent/SocketIntegTests.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.StandardProtocolFamily;
+import java.net.UnixDomainSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.opensearch.test.OpenSearchTestCase;
+
+@SuppressWarnings("removal")
+public class SocketIntegTests extends OpenSearchTestCase {
+
+    private static final List<String> UNIX_OS = List.of("linux", "mac", "unix", "aix", "sunos");
+
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    public void testLocalSocketConnection() throws IOException {
```

Comment:
we can disable these tests until https://github.com/opensearch-project/OpenSearch/issues/17662 is done? i.e we can annotate this with     @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/17662")